src/parser/parse.ts

```diff
@@ -182,6 +189,14 @@ export class Parser<
         }
 
         if (parseFlag(input)) {
+          if (parsingFlags && this.currentFlag && stdinRead) {
```

Comment:
`parseFlag` is responsible for pushing to `this.raw`, so I think it'd be cleaner to move this processing to the `if` block you added on line 160

Reply:
You could make `parseFlag` async. Looks like it's only used once so it should be easy to do